File: Difficulty: Easy/Second Largest/second-largest.py
import heapq
class Solution:
    def getSecondLargest(self, arr):
        # Code Here
        k = 2
        heap = []
        l = list(set(arr))
        if len(l) == 1:
            return -1
        for num in l:
            
            heapq.heappush(heap,num)
            
            if len(heap) > k:
                heapq.heappop(heap)
                
        return heap[0]
                
        
        

File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        // code here
        set<int> s (arr.begin(),arr.end());
        vector<int> v(s.begin(),s.end());
        sort(v.begin(),v.end());
        int n = v.size();
        
        if (v.size()< 2){
            return -1;
        }
        return v[n-2];
        
    }
};